src/silo/query_engine/filter_expressions/insertion_contains.cpp

```diff
@@ -0,0 +1,115 @@
+   Expression::AmbiguityMode /*mode*/
+) const {
+   CHECK_SILO_QUERY(
+      database_partition.columns.insertion_columns.contains(column_name),
+      "The insertion column '" + column_name + "' does not exist."
+   )
+
+   const storage::column::InsertionColumnPartition& insertion_column =
+      database_partition.columns.insertion_columns.at(column_name);
+
+   return std::make_unique<operators::BitmapProducer>(
+      [&]() {
+         auto search_result = insertion_column.search(position, value);
+         return OperatorResult(search_result.release());
+      },
+      database_partition.sequenceCount
+   );
+}
+
+namespace {
+
+std::regex buildValidInsertionSearchRegex() {
+   // Build the following regex pattern: ^([nuc-symbols]|\.\*)*$
+   std::stringstream regex_pattern_string;
+   regex_pattern_string << "^([";
+   for (const auto nuc : NUC_SYMBOLS) {
+      regex_pattern_string << nucleotideSymbolToChar(nuc);
+   }
+   regex_pattern_string << "]|\\.\\*)*$";
+   return std::regex(regex_pattern_string.str());
+}
+
+const std::regex VALID_INSERTION_SEARCH_VALUE_REGEX = buildValidInsertionSearchRegex();
+
+bool validateInsertionSearchValue(const std::string& value) {
+   return std::regex_search(value, VALID_INSERTION_SEARCH_VALUE_REGEX);
+}
+
+}  // namespace
+
+// NOLINTNEXTLINE(readability-identifier-naming)
+void from_json(const nlohmann::json& json, std::unique_ptr<InsertionContains>& filter) {
+   CHECK_SILO_QUERY(
+      json.contains("column"), "The field 'column' is required in an InsertionContains expression"
+   )
+   CHECK_SILO_QUERY(
+      json["column"].is_string(),
+      "The field 'column' in an InsertionContains expression needs to be a string"
+   )
+   CHECK_SILO_QUERY(
+      json.contains("position"),
+      "The field 'position' is required in an InsertionContains expression"
+   )
+   CHECK_SILO_QUERY(
+      json["position"].is_number_unsigned() && (json["position"].get<uint32_t>() > 0),
+      "The field 'position' in an InsertionContains expression needs to be a positive number (> 0)"
+   )
+   CHECK_SILO_QUERY(
+      json.contains("value"), "The field 'value' is required in an InsertionContains expression"
+   )
+   CHECK_SILO_QUERY(
+      json["value"].is_string() && !json["value"].is_null(),
+      "The field 'value' in an InsertionContains expression needs to be a string"
+   )
+   const std::string& column_name = json["column"];
+   const uint32_t position = json["position"];
+   const std::string& value = json["value"].get<std::string>();
+   CHECK_SILO_QUERY(
+      !value.empty(),
+      "The field 'value' in an InsertionContains expression must not be an empty string"
+   )
+   CHECK_SILO_QUERY(
+      validateInsertionSearchValue(value),
+      "The field 'value' in the InsertionContains expression does not contain a valid regex "
+      "pattern: \"" +
+         value + "\". It must only consist of nucleotide symbols and the regex symbol '?'."
```

Comment:
The regex says this:
```suggestion
         value + "\". It must only consist of nucleotide symbols and the regex symbol '.*'."
```

Reply:
https://github.com/GenSpectrum/LAPIS/blob/8de8c7c966ad08cdf9d5ecf5f3baaab060d142e2/server/src/main/antlr/ch/ethz/lapis/api/parser/VariantQuery.g4#L44

As far as I know the user and the query language syntax uses `?` while it then gets parsed to `.*` which is passed on to SILO. But I guess Alex (@Taepper) can better comment on this.

Reply:
SILO should be self-contained and operable without LAPIS. So SILO should return error messages that are valid for SILO.

Reply:
Yes, I would say that SILO should only know about regex. This makes it more flexible to extend or generalise to different columns

Reply:
Alright. Since SILO should only know about regex and be self-contained, I changed the error message from `?` to `.*`